pennylane/drawer/tape_mpl.py

```diff
@@ -325,4 +342,10 @@ def tape_mpl(tape, wire_order=None, show_all_wires=False, decimals=None, **kwarg
                 drawer.measure(n_layers, wire_map[wire])
                 measured_wires += wire
 
+    if style:
+        # we don't want to mess with how it modifies whether the interface is interactive
+        # but we want to restore everything else
+        restore_params["interactive"] = mpl.rcParams["interactive"]
```

Comment:
Excellent find!

Reply:
nice, this was a doozy 😄 
I think this is alright for this PR, but I must confess that I have little faith in matplotlib making our lives easy in the future. We should have a future plan to make this better, or at least some issue to track updating it. In short, I'm thinking that if we know we only change keys {x,y,z}, we should only undo our changes to keys {x,y,z}.

Reply:
This sounds like a `try/finally`, or even `with` use-case?

Reply:
totally slipped my mind, yes! we need to restore this even if something fails.

Reply:
Well, you think the in-build context manager for changing the style `mpl.rc_context` would be able to handle this, but obviously it can't :(

Reply:
wrapping this entire block in a try/finally should do the job, even if a bit ugly from an indentation point of view :)

Reply:
For now though, I think this implementation works fine.